src/code_matcher.ml

```diff
@@ -131,7 +133,9 @@ struct
         let y = remove_loc y in
         if Poly.( <> ) x y then (
           let round_trip =
-            remove_loc (parse_string (Format.asprintf "%a@." M.pp x))
+            let compiler_x = M.to_compiler x in
```

Comment:
Could we leave a comment here for the future ppxlib devs about why exactly we do this -- something like your PR comment would be great?

Reply:
Sure thing!